hibernate-vector/src/main/java/org/hibernate/vector/AbstractOracleVectorJdbcType.java

```diff
@@ -0,0 +1,177 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.OracleDialect;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.descriptor.ValueBinder;
+import org.hibernate.type.descriptor.ValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.BasicPluralJavaType;
+import org.hibernate.type.descriptor.java.ByteJavaType;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.java.PrimitiveByteArrayJavaType;
+import org.hibernate.type.descriptor.jdbc.ArrayJdbcType;
+import org.hibernate.type.descriptor.jdbc.BasicBinder;
+import org.hibernate.type.descriptor.jdbc.BasicExtractor;
+import org.hibernate.type.descriptor.jdbc.JdbcLiteralFormatter;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.hibernate.type.descriptor.jdbc.internal.JdbcLiteralFormatterArray;
+
+import oracle.jdbc.OracleType;
+
+/**
+ * Specialized type mapping for generic vector {@link SqlTypes#VECTOR} SQL data type for Oracle.
+ * <p>
+ * This class handles generic vectors represented by an asterisk (*) in the format,
+ * allowing for different element types within the vector.
+ *
+ * @author Hassan AL Meftah
+ */
+public abstract class AbstractOracleVectorJdbcType extends ArrayJdbcType {
+
+	protected Class<?> javaType;
+
+	public AbstractOracleVectorJdbcType(JdbcType elementJdbcType, Class<?> javaType) {
+		super( elementJdbcType );
+		this.javaType = javaType;
+	}
+
+	public abstract String getFriendlyName();
+
+	public abstract void appendWriteExpression(String writeExpression, SqlAppender appender, Dialect dialect);
+
+	@Override
+	public int getJdbcTypeCode() {
+		return SqlTypes.VECTOR;
+	}
+
+	@Override
+	public <T> JdbcLiteralFormatter<T> getJdbcLiteralFormatter(JavaType<T> javaTypeDescriptor) {
+		final JavaType<T> elementJavaType;
+		if ( javaTypeDescriptor instanceof PrimitiveByteArrayJavaType ) {
+			// Special handling needed for Byte[], because that would conflict with the VARBINARY mapping
+			//noinspection unchecked
+			elementJavaType = (JavaType<T>) ByteJavaType.INSTANCE;
+		}
+		else if ( javaTypeDescriptor instanceof BasicPluralJavaType ) {
+			//noinspection unchecked
+			elementJavaType = ( (BasicPluralJavaType<T>) javaTypeDescriptor ).getElementJavaType();
+		}
+		else {
+			throw new IllegalArgumentException( "not a BasicPluralJavaType" );
+		}
+		return new JdbcLiteralFormatterArray<>(
+				javaTypeDescriptor,
+				getElementJdbcType().getJdbcLiteralFormatter( elementJavaType )
+		);
+	}
+
+	@Override
+	public String toString() {
+		return "OracleVectorTypeDescriptor";
+	}
+
+	/**
+	 * Checks if the Oracle JDBC driver version is less than 23.4.
+	 *
+	 * @return true if the driver version is less than 23.4, false otherwise
+	 */
+	protected boolean isVectorSupportedByDriver(WrapperOptions options) throws SQLException {
```

Comment:
Instead of introducing a method that is repeatedly called, this is something that should be passed a boolean flag to the constructor of the various `JdbcType` classes.